incubator/hnc/internal/reconcilers/object.go

```diff
@@ -699,6 +705,21 @@ func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstruct
 	}
 }
 
+// getNoneSelector returns true indicates that user do not want this object to be propagated
+func (r *ObjectReconciler) getNoneSelector(log logr.Logger, inst *unstructured.Unstructured) bool {
+	annot := inst.GetAnnotations()
+	noneSelectorStr, ok := annot[api.AnnotationNoneSelector]
+	if !ok {
+		return false
+	}
+	noneSelector, err := strconv.ParseBool(noneSelectorStr)
+	if err != nil {
+		log.Error(err, "Invalid noneSelector value, it should be either true or false.")
+		return false
+	}
+	return noneSelector
```

Comment:
Remind me if we agreed on this? Another option here is "any non-empty string means `true`, empty string or missing means `false`." I'm not sure what the K8s standard is though.

Reply:
Here also includes the case where user inputs an invalid value (i.e. neither "true" or "false"), and I don't think these should be allowed.

For the empty string, do we want to treat it as `false` or not-allowed?

Reply:
Ok maybe let's keep this as-is. At least it's clear :)